specification/hugr.md

```diff
@@ -24,61 +24,62 @@ represent (typed) data or control dependencies.
-    programs and programs from future front-ends designed to target
-    HUGR. These including the upcoming Python eDSL for quantum-classical
-    programming, and BRAT (which already uses an internal graph-like
-    representation for classical functional programs and quantum
-    kernels). We expect that these front-ends will provide
-    programmer-facing control flow constructs that map to the preferred
-    constructs in HUGR without first having to pass through an
-    LLVM/SSACFG intermediate.
-  - Support for nested structures. The nodes form a tree-like hierarchy
-    with nested graphs encoded as children of their containing node.
-  - User-defined metadata, such as debug information, can be efficiently
-    attached to nodes and queried.
-  - All user-provided information can be encoded and decoded in a stable
-    (versioned) efficient serialized format.
-  - A type system for checking valid operation connectivity + (nice to
-    have) only operations supported on specific targets are used.
-  - A space efficient and user friendly specification of a subgraph and
-    replacement graph, along with an efficient routine for performing
-    the replacement.
+- A directed graph structure with extensible operation types in the
+  nodes and data types in the edges.
+- Indexed connection ports for each operation node, which may be
+  connected to another port with the same data type or remain
+  unconnected.
+- Control-flow support with ability to capture both LLVM SSACFG style
+  programs and programs from future front-ends designed to target
+  HUGR. These include the [guppylang](https://github.com/CQCL/guppylang)
+  Python eDSL for quantum-classical programming,
+  and BRAT (which already uses an internal graph-like
+  representation for classical functional programs and quantum
+  kernels). We expect that these front-ends will provide
+  programmer-facing control flow constructs that map to the preferred
+  constructs in HUGR without first having to pass through an
+  LLVM/SSACFG intermediate.
+- Support for nested structures. The nodes form a tree-like hierarchy
+  with nested graphs encoded as children of their containing node.
+- User-defined metadata, such as debug information, can be efficiently
+  attached to nodes and queried.
+- All user-provided information can be encoded and decoded in a stable
+  (versioned) efficient serialized format.
+- A type system for checking valid operation connectivity + (nice to
+  have) only operations supported on specific targets are used.
+- A space efficient and user friendly specification of a subgraph and
+  replacement graph, along with an efficient routine for performing
+  the replacement.
 
 ## Functional description
 
-A HUGR is a directed graph. There are several different types of node, and
-several different types of edge, with different semantics, described below.
+A HUGR is a directed graph. There are several different types of nodes, and
```

Comment:
There is a lot of discussion on this at https://english.stackexchange.com/questions/29843/types-of-followed-by-singular-or-plural but since it's a matter of preference, I have reverted my changes.

Reply:
I wonder if there is a difference between American and British English here. I was interested in [this comment](https://english.stackexchange.com/a/5546) (from a linked page):

> I think that the construction of the form "two types of apple" sounds more than stilted; it just plain sounds awkward, and I would be surprised that it sounds familiar and normal to anyone (at least speaking for US English).

To me it sounds perfectly natural!

Reply:
Most likely, as I have the same experience as that commenter.